flutter-studio/src/io/flutter/module/FlutterModuleImporter.java

```diff
@@ -98,18 +101,23 @@ public void importModule() {
     // After both buffers are full, write them both to their original files.
     ApplicationManager.getApplication().runWriteAction(() -> {
       try {
-        settingsFile.setBinaryContent(settingsWriter.toString().getBytes(Charset.defaultCharset()));
-        buildFile.setBinaryContent(buildWriter.toString().getBytes(Charset.defaultCharset()));
+        settingsFile.setBinaryContent(settingsWriter.toString().getBytes(StandardCharsets.UTF_8));
+        buildFile.setBinaryContent(buildWriter.toString().getBytes(StandardCharsets.UTF_8));
       }
       catch (IOException e) {
         // Should not happen
       }
     });
+
+    // Setup a default run configuration for 'main.dart' (if it's not there already and the file exists).
+    PubRoot pubRoot = PubRoot.forDirectory(moduleRoot);
+    assert(pubRoot != null);
```

Comment:
How similar is this code path to the one in `FlutterBuildActionGroup.build()`? I ask because we (very infrequently) see an NPE from there when the pubRoot is null. https://github.com/flutter/flutter-intellij/issues/4065

Reply:
If it's possible here as well, we'd want to handle null instead of asserting.

Reply:
I don't see any similarity. Looking at the caller of `FlutterBuildActionGroup.build()` I see it does `PubRoot.forEventWithRefresh(event)`. My guess is the null arises because the event changes status before that expression runs, and the Project cannot be retrieved from the event. All I have is anecdotal evidence for this, but I think JetBrains is tightening controls on how much can be done on the EDT, probably to improve responsiveness. We should look at callers of `forEventWithRefresh` and rewrite them to ensure the Project is fetched early.